pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -121,28 +121,35 @@ func (r *ReconcileUserAccount) Reconcile(request reconcile.Request) (reconcile.R
 	// If the UserAccount has not been deleted, create or update user and identity resources.
 	// If the UserAccount has been deleted, delete secondary resources identity and user.
 	if !util.IsBeingDeleted(userAcc) {
+		reqLogger.Info("Adding finalizer on UserAccount")
 		// Add the finalizer if it is not present
 		if err := r.addFinalizer(userAcc); err != nil {
 			return reconcile.Result{}, err
 		}
-
+	}
+	if !util.IsBeingDeleted(userAcc) && !userAcc.Spec.Disabled {
+		reqLogger.Info("Ensuring user and identity associated with UserAccount")
 		var createdOrUpdated bool
 		var user *userv1.User
 		if user, createdOrUpdated, err = r.ensureUser(reqLogger, userAcc); err != nil || createdOrUpdated {
 			return reconcile.Result{}, err
 		}
-
 		if _, createdOrUpdated, err = r.ensureIdentity(reqLogger, userAcc, user); err != nil || createdOrUpdated {
 			return reconcile.Result{}, err
 		}
-
 		if _, createdOrUpdated, err = r.ensureNSTemplateSet(reqLogger, userAcc); err != nil || createdOrUpdated {
 			return reconcile.Result{}, err
 		}
-	} else if util.HasFinalizer(userAcc, userAccFinalizerName) {
-		if err = r.manageCleanUp(userAcc); err != nil {
+	} else if util.HasFinalizer(userAcc, userAccFinalizerName) || userAcc.Spec.Disabled {
+		reqLogger.Info("Deleting user and identity associated with UserAccount")
+		if err = r.deleteUserIdentity(userAcc); err != nil {
 			return reconcile.Result{}, err
 		}
+
+		if userAcc.Spec.Disabled {
```

Comment:
As Shane and me already pointed we should not set "disabled" status in the same reconcile loop as the user/identity deletion. `deleteUserIdentity(userAcc)` should return a bool (see https://github.com/codeready-toolchain/member-operator/pull/105#discussion_r368707702). Let's say it returns `true` if the user OR the identity were actually deleted. If both User and Identity do not exist (so nothing to delete) then it returns `false` (nothing deleted). Similar to what we do with ensure methods.
Then we should update the status to "disabled" only if `deleteUserIdentity()` returns false. If it returns `true` (and spec.disabled == true) we could even set the status to "disabling". It will guarantee us that we update the status after the deletion. Not when we just requested it.

Reply:
fixed!